submissions/ArthurGorbenko/Friend's app/js/script.js

```diff
@@ -0,0 +1,256 @@
+
+  append(CONSTS.LIST_OF_CARDS, li);
+};
+
+const refreshButtons = () => {
+  CONSTS.LIST_OF_CARDS.addEventListener("click", writeLetter);
+};
+
+const makeDarkBackground = () => {
+  CONSTS.BODY.classList.toggle("bodyDark");
+  CONSTS.MENU.classList.toggle("menuDark");
+  CONSTS.DM_FORM.classList.toggle("directMessageDark");
+};
+
+const changeColorCards = () => {
+  for (let key = 0; key < CONSTS.CARDS.length; key++) {
+    CONSTS.CARDS[key].classList.toggle("cardDark");
+  }
+  for (let index = 0; index < CONSTS.BUTTONS.length; index++) {
+    CONSTS.BUTTONS[index].classList.toggle("buttonDark");
+  }
+};
+
+const createNewList = people => {
+  CONSTS.LIST_OF_CARDS.innerHTML = "";
+  people.forEach(man => createCard(man));
+  if (CONSTS.DARK_INPUT.checked) {
+    for (let key = 0; key < CONSTS.CARDS.length; key++) {
+      CONSTS.CARDS[key].classList.add("cardDark");
+    }
+    for (let index = 0; index < CONSTS.BUTTONS.length; index++) {
+      CONSTS.BUTTONS[index].classList.add("buttonDark");
+    }
+  }
+  refreshButtons();
+};
+
+CONSTS.SEND_BUTTON.addEventListener("click", messageSend);
+
+const processCards = subString => {
+  if (!subString || subString === " ") {
+    return null;
+  }
+  if (!FILTERED_BY_GENDER) {
+    users = usersConst;
+  }
+  subString = subString.toLowerCase();
+  let usersFound = users.filter(
+    user => user.name.first.match(subString) || user.name.last.match(subString)
+  );
+  createNewList(usersFound);
+};
+
+let query = "";
+
+CONSTS.SEARCH_FIELD.addEventListener("keyup", function() {
+  query = this.value.toLowerCase();
+  processCards(query);
+});
+
+const showGender = genderType => {
+  if (FILTERED_BY_GENDER) {
+    users = usersConst;
+  }
+  users = users.filter(user => user.gender === genderType);
+  createNewList(users);
+  FILTERED_BY_GENDER = true;
+  if (query) {
+    processCards(query);
+  }
+};
+
+const genderFilter = ({ target }) => {
+  if (target.checked) {
+    if (target.dataset.gender === "male") {
+      showGender(target.dataset.gender);
+    } else if (target.dataset.gender === "female") {
+      showGender(target.dataset.gender);
+    }
+  }
+};
+
+CONSTS.RADIO_GENDER_FORM.addEventListener("change", genderFilter);
+
+const sortingProcess = typeOfSort => {
+  const runSort = (a, b) => a.dob.age - b.dob.age;
+  if (typeOfSort === "ascending") {
+    users.sort(runSort);
+  } else {
+    users.sort((a, b) => runSort(b, a));
+  }
+  createNewList(users);
+  processCards(query);
+};
+
+const ageSort = ({ target }) => {
+  if (target.checked) {
+    if (target.dataset.render === "descending") {
+      sortingProcess("descending");
+    } else {
+      sortingProcess("ascending");
+    }
+  }
+};
+
+CONSTS.RADIO_AGE_FORM.addEventListener("change", ageSort);
+
+const resetRadio = allButtons => {
+  for (let key = 0; key < allButtons.length; key++) {
+    allButtons[key].checked = false;
+  }
+};
+
+const resetAll = () => {
```

Comment:
Do not fear make longer name. Be more specific about what function will do. 

Reply:
Rename.